Several test cases failing on live server due to a login issue

Several, but not all, are failing to a login page issue. Here is an example,

java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at teammates.test.pageobjects.AppPage.createNewPage(AppPage.java:1128)
	at teammates.test.pageobjects.AppPage.getNewPageInstance(AppPage.java:160)
	at teammates.test.cases.ui.browsertests.StudentFeedbackSubmitPageUiTest.loginToStudentFeedbackSubmitPage(StudentFeedbackSubmitPageUiTest.java:518)
	at teammates.test.cases.ui.browsertests.StudentFeedbackSubmitPageUiTest.testSubmitAction(StudentFeedbackSubmitPageUiTest.java:368)
	at teammates.test.cases.ui.browsertests.StudentFeedbackSubmitPageUiTest.testAll(StudentFeedbackSubmitPageUiTest.java:59)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:84)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:714)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:901)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1231)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:127)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:111)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:334)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:329)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:291)
	at org.testng.SuiteRunner.run(SuiteRunner.java:240)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1149)
	at org.testng.TestNG.run(TestNG.java:1057)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:111)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:204)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:175)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at teammates.test.pageobjects.AppPage.createNewPage(AppPage.java:1124)
	... 28 more
Caused by: java.lang.IllegalStateException: Not in the correct page!
	at teammates.test.pageobjects.AppPage.<init>(AppPage.java:142)
	at teammates.test.pageobjects.FeedbackSubmitPage.<init>(FeedbackSubmitPage.java:13)
	... 33 more

--- a/src/test/java/teammates/test/cases/ui/browsertests/StudentFeedbackSubmitPageUiTest.java
+++ b/src/test/java/teammates/test/cases/ui/browsertests/StudentFeedbackSubmitPageUiTest.java
@@ -364,7 +364,7 @@ public class StudentFeedbackSubmitPageUiTest extends BaseUiTestCase {
         submitPage.verifyHtmlMainContent("/studentFeedbackSubmitPageFullyFilled.html");
 
         ______TS("create new response for unreg student");
-
+        submitPage.logout();
         submitPage = loginToStudentFeedbackSubmitPage(testData.students.get("DropOut"), "Open Session");
 
         submitPage.fillResponseTextBox(1, 0, "Test Self Feedback");

--- a/src/test/java/teammates/test/pageobjects/AppPage.java
+++ b/src/test/java/teammates/test/pageobjects/AppPage.java
@@ -396,6 +396,7 @@ public abstract class AppPage {
      */
     public static void logout(Browser currentBrowser){
         currentBrowser.driver.get(TestProperties.inst().TEAMMATES_URL + Const.ViewURIs.LOGOUT);
+        currentBrowser.selenium.waitForPageToLoad(TestProperties.inst().TEST_TIMEOUT_PAGELOAD);
         currentBrowser.isAdminLoggedIn = false;
     }
     

